test/SILOptimizer/return.swift

```diff
@@ -97,7 +97,7 @@ func whileLoop(flag: Bool) -> Int {
 struct S {}
 extension S:ExpressibleByStringLiteral {
   init!(stringLiteral:String) {
-  } // expected-error {{missing return in a function expected to return 'S!'}}
+  } // no error
```

Comment:
@graydon This error is printed since your fix https://github.com/apple/swift/pull/11765
But I think the error message was wrong. Please confirm.

Reply:
@eeckstein Perhaps! I have no idea whether an IUO initializer with no body is legal, I was mainly concerned with making it stop crashing in #11765 by handling the case, and the existing diagnostic seemed to fit my understanding of the meaning of an IUO initializer with no body. But by all means confirm with someone who knows IUO semantics better than me.

Reply:
Returning successfully (i.e. a non-optional, properly-initialized `S`) would be the correct semantics.

Reply:
Yes, it returns an S (implicitly, not with a return statement)